scholia/app/templates/author_timeline.sparql

```diff
@@ -1,108 +1,133 @@
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX wdt: <http://www.wikidata.org/prop/direct/>
+SELECT ?label ?beginning ?ending ?education_degree_label WHERE {
   {
-    { 
+    {
       SELECT ?work ?beginning WHERE {
-        INCLUDE %works
+        {
+          SELECT ?work ?beginning (COUNT(?citing_article) AS ?number_of_citations) WHERE {
+            {
+              SELECT ?work (MIN(?publication_date) AS ?beginning) WHERE {
+                ?work wdt:P50 target: .
+                ?work wdt:P577 ?publication_date .
+              }
+              GROUP BY ?work
+            }
+            OPTIONAL {
+              ?citing_article wdt:P2860 ?work .
+            }
+          }
+          GROUP BY ?work ?beginning
+        }
       }
       ORDER BY DESC(?number_of_citations)
-      LIMIT 1        
+      LIMIT 1
     }
     BIND("📖 publication of most cited article" AS ?label)
   }
-  UNION
-  {
-    { 
+  UNION {
+    {
       SELECT ?work ?beginning WHERE {
-        INCLUDE %works
+        {
+          SELECT ?work ?beginning (COUNT(?citing_article) AS ?number_of_citations) WHERE {
+            {
+              SELECT ?work (MIN(?publication_date) AS ?beginning) WHERE {
+                ?work wdt:P50 target: .
+                ?work wdt:P577 ?publication_date .
+              }
+              GROUP BY ?work
+            }
+            OPTIONAL {
+              ?citing_article wdt:P2860 ?work .
+            }
+          }
+          GROUP BY ?work ?beginning
+        }
       }
       ORDER BY ?beginning
-      LIMIT 1        
+      LIMIT 1
     }
     BIND("📖 first publication" AS ?label)
   }
-  UNION
-  {
-    { 
+  UNION {
+    {
       SELECT ?work ?beginning WHERE {
-        INCLUDE %works
+        {
+          SELECT ?work ?beginning (COUNT(?citing_article) AS ?number_of_citations) WHERE {
+            {
+              SELECT ?work (MIN(?publication_date) AS ?beginning) WHERE {
+                ?work wdt:P50 target: .
+                ?work wdt:P577 ?publication_date .
+              }
+              GROUP BY ?work
+            }
+            OPTIONAL {
+              ?citing_article wdt:P2860 ?work .
+            }
+          }
+          GROUP BY ?work ?beginning
+        }
       }
       ORDER BY DESC(?beginning)
-      LIMIT 1        
+      LIMIT 1
     }
     BIND("📖 latest publication" AS ?label)
   }
-  UNION
-  {
+  UNION {
     # Education
     target: p:P69 ?education_statement .
     ?education_statement ps:P69 ?education .
-    ?education rdfs:label ?education_label . 
-    FILTER (lang(?education_label) = 'en')
-    BIND(CONCAT("🦉 ", ?education_label) AS ?label)
-    OPTIONAL { ?education_statement pq:P580 ?beginning . }
-    OPTIONAL { ?education_statement pq:P582 ?ending . }
+    ?education rdfs:label ?education_label_raw .
+    FILTER(LANG(?education_label_raw) = "en")
+    OPTIONAL {
+      ?education_statement pq:P580 ?beginning .
+    }
     OPTIONAL {
```

Comment:
Please add `target: p:P69 ?education_statement .` to this OPTIONAL, which should make it faster.